### PR TITLE
rename jshintAddonTree to linterAddonTree

### PR DESCRIPTION
**Pull request by billybonks**
While investigating something else i noticed that this method name didn't really describe what the function was doing any more :)

### CODE REVIEW

**Comment by Turbo87** [comment]
looks like CI is failing 🤔

**Comment by kellyselden** [comment]
@rwjblue Correct me if I'm wrong, but even though this is labelled as private, this is probably going to break people using these functions. We should probably have a deprecation period.

**Comment by rwjblue** [comment]
>   Correct me if I'm wrong, but even though this is labelled as private, this is probably going to break people using these functions. We should probably have a deprecation period.

Correct, but it would have to be someone directly calling `jshintAddonTree` which seems pretty unusual (there are only two usages on [emberobserver.com code search](https://emberobserver.com/code-search?codeQuery=jshintAddonTree)).  Most folks would rely on the linting behavior implicitly, so they wouldn't notice the rename. I'm still 👍 on a deprecation though...

File: lib/models/addon.js [hunk 1]
```diff
@@ -1179,13 +1179,13 @@ let addonProto = {
...
+    Invokes lintTree for addon that implements lintTree function
 
     @private
-    @method jshintAddonTree
-    @return {Tree} Tree with JShint output (tests)
+    @method linterAddonTree
+    @return {Tree} Tree with tests generated by included linters
   */
-  jshintAddonTree() {
+  linterAddonTree() {
```
Comment (rwjblue) [inline lib/models/addon.js]: Lets call it `lintAddonTree` instead

**Comment by billybonks** [comment]
deprecated original function / renamed rename 😄 looking into cli, let me know about the deprecation copy.

**Comment by billybonks** [comment]
it seems like master is failing in general on same spec https://travis-ci.org/ember-cli/ember-cli/jobs/360034648#L1969

File: lib/models/addon.js [hunk 2]
```diff
@@ -1239,7 +1243,7 @@ let addonProto = {
     return new Funnel(lintedAddon, {
       srcDir: '/',
       destDir: `${this.name}/tests/`,
-      annotation: `Funnel: Addon#jshintAddonTree(${this.name})`,
+      annotation: `Funnel: Addon#lintAddonTree(${this.name})`,
```
Comment (billybonks) [inline lib/models/addon.js]: I am not sure if stats are depending on annotations, so the question is that do we need a long lead time to change the annotation.

**Comment by rwjblue** [comment]
Sorry for the delay here, but I think its probably best to close this PR. I think that we should _probably_ consider deprecating the tree based linting hooks (we've migrated default apps off of them as of 3.17.0-beta.1).

**Comment by gabrielcsapo** [comment]
is there an open issue to remove jshintAddonTree?

**Comment by rwjblue** [comment]
Dunno, I don't see an obvious one...

**Pull request closed without merging**